### PR TITLE
keep the trailing slash in the path for regexp matching

### PR DESCRIPTION
**Pull request by aryszka**
Fixing: #1210 

Signed-off-by: Arpad Ryszka <arpad.ryszka@gmail.com>

### CODE REVIEW

**Comment by NorthFury** [comment]
:+1:

File: routing/matcher.go [hunk 1]
```diff
@@ -326,7 +315,8 @@ func newMatcher(rs []*Route, o MatchingOptions) (*matcher, []*definitionError) {
 		}
 
 		if r.pathSubtree != "" {
-			path, _ := cleanPath(r.pathSubtree, o|IgnoreTrailingSlash)
+			path := httppath.Clean(r.pathSubtree)
+			path = trimTrailingSlash(path)
```
Comment (aryszka) [inline routing/matcher.go]: don't we need to use the `if o.ignoreTrailingSlash()` condition here?

Comment (aryszka) [reply routing/matcher.go]: oops.. my bad for not watching :man_facepalming:

**Comment by aryszka** [comment]
:+1:

File: routing/matcher.go (hunk 1 above)
Comment (NorthFury) [reply routing/matcher.go]: `(o|IgnoreTrailingSlash).ignoreTrailingSlash()` is always true

**Comment by szuecs** [comment]
:+1:

**Pull request merged**